controllers/storageautoscaler/storageautoscaler_controller.go

```diff
@@ -67,6 +67,32 @@ func (r *StorageAutoscalerReconciler) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	// if more than one storage autoscaler is present in the cluster with same device class and same storage cluster
```

Comment:
Could factor this error check to a helper method? Really long methods are more difficult to maintain in the long term.

Reply:
DOne